---
title: "Somascan_Analysis"
format: html
editor: visual
---

# Read in the adat file and manipulate the dataframe

```{r}
library("SomaDataIO")

master <- read_adat("~/Documents/R scripts/Soma_project/L0125004423_v5.0_EDTAPlasma.hybNorm.medNormInt.plateScale.calibrate.anmlQC.qcCheck.anmlSMP.20251008.adat")

head(master)

```

## Manipulate the columns

```{r}
library(dplyr)

#remove calibrators and flagged samples
df <- master %>%
  filter(
    !SampleType %in% c("Calibrator", "Buffer", "QC"),
    RowCheck != "FLAG"
  ) %>% as_tibble()

#remove unwanted columns

df <- df %>% dplyr::select(-PlateId,-PlateRunDate, -ScannerID, -PlatePosition, -SlideId, -Subarray,-PercentDilution,-SampleMatrix,-Barcode,-Barcode2d,-SampleNotes,-AliquotingNotes, -SampleDescription, -AssayNotes, -ExtIdentifier, -SsfExtId,-SampleGroup,-SubjectID, -CLI, -RMA, -HybControlNormScale,-NormScale_20, -NormScale_0_005, -NormScale_0_5, -ANMLFractionUsed_20, -ANMLFractionUsed_0_005, -ANMLFractionUsed_0_5,-SiteId,-TimePoint, -SampleName, -SampleType, -RowCheck)

rownames(df) = NULL

```

## convert seq IDs to Uniprot

```{r}
library(stringr)
library(readr)
library(dplyr)

#load in the conversion key csv
map <- read.csv("seqid_to_uniprot.csv")

#reformatting (eg. 10023-32 to seq.10023.32)
map <- map %>% mutate(
  seqid_formatted = paste0("seq.", gsub("-",".",SeqId))
)

# make the vector from the map df
id_map <- setNames(map$UniProtID, map$seqid_formatted)

# don't rename the seqID column
old_names <- colnames(df)
new_names <- old_names
new_names[-1] <- ifelse(
  old_names[-1] %in% names(id_map),
  id_map[old_names[-1]],
  old_names[-1]
)

colnames(df) <- new_names

#remove any columns that still start with "seq" because these are controls

names(df) <- make.unique(names(df)) # some somamer sequences map to the same uniprotID but different regimes of the protein(cytoplasmic vs extracellular, etc. this makes the column names unique to avoid errors on dplyr.)

df <- df %>% dplyr::select(-starts_with("seq"))

```

## Add a Timepoint and Subject ID column

```{r}
df <- df %>%
  mutate(
    Timepoint = str_extract(SampleId, "D[1-5]$"),       # extract D1–D5 at the end
    SubjectID = str_remove(SampleId, "D[1-5]$")         # remove D1–D5 to get the subject ID
  ) %>%
  relocate(SubjectID, .before = SampleId) %>%            # move SubjectID before SampleId
  relocate(Timepoint, .after = SampleId)

df$Timepoint <- factor(df$Timepoint)

df <- as_tibble(df)
```

## log2 transformation of the proteomics data

```{r}

df_log2 <- df %>% mutate(across(where(is.numeric), ~ log2(.x)))

```

# Plot analytes of interest

```{r}
library(tidyverse)

# Vector of target proteins
target_proteins <- c("P05062", "P15559", "P21695")

# Convert to long format for ggplot
df_long <- df %>%
  pivot_longer(cols = all_of(target_proteins),
               names_to = "Protein",
               values_to = "Expression")

# Summary stats: mean and SE per group
summary_df <- df_long %>%
  group_by(Protein, Timepoint) %>%
  summarise(
    mean_expr = mean(Expression, na.rm = TRUE),
    sd_expr = sd(Expression, na.rm = TRUE),
    .groups = "drop"
  )

# Plot
plot1 <- ggplot(summary_df, aes(x = Timepoint, y = mean_expr, fill = Timepoint)) +
  geom_col(position = "dodge", color = "black", width = 0.8) +
  geom_errorbar(aes(ymin = mean_expr - sd_expr, ymax = mean_expr + sd_expr),
                width = 0.2, position = position_dodge(0.7)) +
  facet_wrap(~ Protein, scales = "free_y", ncol = 4) +
  theme_minimal(base_size = 14) +
  theme(
    panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(),
    strip.text = element_text(face = "bold", size = 13, margin = margin(b=20)),
    axis.text.x = element_text(angle = 45, hjust = 1),
    legend.position = "none"
  ) +
  labs(
    title = "",
    x = "Draw",
    y = "Mean Expression"
  )
```


# sPLSDA

```{r}
library(mixOmics)
library(tidyverse)

set.seed(123)
plsda_matrix <- df_log2 %>% dplyr::select((where(is.numeric)))
plsda_response <- df_log2$Timepoint
# run the PLSDA

plsda_result <- splsda(plsda_matrix, plsda_response, ncomp = 3, keepX = c(75,75,75), scale = T) #z scores the data within the plsda
PLS_scores <- as.data.frame(plsda_result$variates$X)
PLS_scores$Response <- df_log2$Timepoint

# plot the result

explained_var <- plsda_result$prop_expl_var$X
x_lab <- paste0("Component 1 (", round(explained_var[1] * 100, 1), "%)")
y_lab <- paste0("Component 2 (", round(explained_var[2] * 100, 1), "%)")


plsda_plot <- ggplot(PLS_scores, aes(x = comp1, y = comp2, color = Response, fill = Response)) +
  geom_point(size = 3, alpha = 0.8) +
  stat_ellipse(type = "norm", level = 0.85, geom = "polygon", alpha = 0.2, color = NA)+
  scale_fill_discrete(guide = "none") +
  theme_minimal(base_size = 14) +
  labs(
    title = "Proteins over Time",
    x = x_lab,
    y = y_lab,
    color = "Blood Draw Timepoint",
  ) +
  theme(
    plot.title = element_text(hjust = 0.5),
    legend.position = "right",
    axis.text = element_text(size = 14)
  )

loadings <- as.data.frame(plsda_result$loadings$X) %>%
  rownames_to_column(var="Protein")

top20_loadings <- loadings %>%
  mutate(abs_loading = abs(comp1)) %>%
  arrange(desc(abs_loading)) %>%
  dplyr::slice(1:20)

loadings_plot <- ggplot(top20_loadings, aes(x = reorder(Protein, comp1), y = comp1, fill = comp1 > 0)) +
  geom_col(show.legend = FALSE) +
  scale_fill_manual(values = c("indianred3", "steelblue")) +
  coord_flip() +
  labs(
    title = "Top 20 PLS Loadings (C2)",
    x = "Protein",
    y = "Loading Value"
  ) +
  theme_minimal()
#dir.create("PLSDA_results", showWarnings = FALSE)
write_csv(loadings, "PLSDA_results/PLSDA_loadings.csv")

```


## Stratified Bootstrapping for feature stability analysis

```{r}
library(mixOmics)
library(tidyverse)

set.seed(123)

# this performs stratified resampling (subject-level bootstrapping), stores the variables from each component in each iteration, and then reports the frequencies of the variables in each model. Finally it selects variables per component whose frequency is > 0.4.


X <- df_log2 %>% dplyr::select(where(is.numeric)) %>% as.matrix() # define the matrix
Y <- df_log2$Timepoint
SubjectID <- df_log2$SubjectID

nboot <- 500 # number of iterations
ncomp <- 3
keepX <- c(75, 75, 75)


# cluster bootstrap: sample SUBJECTS with replacement; keep all rows per subject
cluster_boot_idx <- function(subject_id) {
  subj <- unique(subject_id)
  boot_subj <- sample(subj, length(subj), replace = TRUE)
  unlist(lapply(boot_subj, function(s) which(subject_id == s)), use.names = FALSE)
}

# store selected variables per component across bootstraps
selected_vars <- vector("list", ncomp)
for (k in seq_len(ncomp)) selected_vars[[k]] <- character(0)

# ---- bootstrap loop ----
for (b in seq_len(nboot)) {

  idx <- cluster_boot_idx(SubjectID)

  fit <- splsda(
    X[idx, ],
    Y[idx],
    ncomp = ncomp,
    keepX = keepX,
    scale = TRUE
  )

  for (k in seq_len(ncomp)) {
    selected_vars[[k]] <- c(selected_vars[[k]], selectVar(fit, comp = k)$name)
  }
}

# ---- variables per component with selection frequency ----
freq_tables <- lapply(seq_len(ncomp), function(k) {
  tibble(Protein = selected_vars[[k]]) %>%
    count(Protein) %>%
    mutate(freq = n / nboot) %>%
    arrange(desc(freq))
})
names(freq_tables) <- paste0("Comp", seq_len(ncomp))

stable_vars_04 <- lapply(freq_tables, function(df) {
  df %>%
    filter(freq > 0.4) %>%
    pull(Protein)
})


```

# GSEA and Over-representation Analysis

-   starting with GSEA: get a list of log2 fold changes between groups of interest, as defined by the groups that clustered separately in PLSDA

-   use the log2fold change to perform GO:BP

## GSEA

### Get the log2fold changes

```{r}
library(tidyverse)
library(clusterProfiler)
library(enrichplot)

# compute log2foldchange of D1 vs D2 and D2 vs D3

# D1 vs D2
fc_df_D1D2 <- df_log2 %>%
  filter(Timepoint %in% c("D1", "D2")) %>%
  pivot_longer(
    cols = -c(SubjectID, SampleId, Timepoint),
    names_to = "Protein",
    values_to = "log2_expr"
  ) %>%
  pivot_wider(
    id_cols    = c(SubjectID, Protein),
    names_from = Timepoint,
    values_from = log2_expr
  ) %>%
  filter(!is.na(D1) & !is.na(D2)) %>%
  mutate(log2_FC_D2_vs_D1 = D2 - D1)

         
ranks_df_D1D2 <- fc_df_D1D2 %>%
  group_by(Protein) %>%
  summarize(meanlog2fc = mean(log2_FC_D2_vs_D1, na.rm = TRUE), .groups = "drop")

# strip probe suffix like ".1" or ".2" from UniProt IDs. This is an artifact of multiple somamer seqIDs mapping to a single protein.
ranks_df_D1D2 <- ranks_df_D1D2 %>%
  mutate(
    UNIPROT_raw = Protein,
    UNIPROT = str_replace(Protein, "\\.\\d+$", ""),   # removes trailing .<digits>
    meanlog2fc = as.numeric(meanlog2fc)
  )

# Collapse duplicates created by stripping suffix. choose the probe with the largest absolute effect size. do not average.

ranks_df_D1D2 <- ranks_df_D1D2 %>%
  group_by(UNIPROT) %>%
  slice_max(order_by = abs(meanlog2fc), n = 1, with_ties = FALSE) %>%
  ungroup()


#D2 vs D3
fc_df_D2D3 <- df_log2 %>%
  filter(Timepoint %in% c("D2", "D3")) %>%
  pivot_longer(
    cols = -c(SubjectID, SampleId, Timepoint),
    names_to = "Protein",
    values_to = "log2_expr"
  ) %>%
  pivot_wider(
    id_cols    = c(SubjectID, Protein),
    names_from = Timepoint,
    values_from = log2_expr
  ) %>%
  filter(!is.na(D2) & !is.na(D3)) %>%
  mutate(log2_FC_D3_vs_D2 = D3 - D2)

ranks_df_D2D3 <- fc_df_D2D3 %>%
  group_by(Protein) %>%
  summarize(meanlog2fc = mean(log2_FC_D3_vs_D2, na.rm = TRUE), .groups = "drop")

ranks_df_D2D3 <- ranks_df_D2D3 %>%
  mutate(
    UNIPROT_raw = Protein,
    UNIPROT = str_replace(Protein, "\\.\\d+$", ""),   # removes trailing .<digits>
    meanlog2fc = as.numeric(meanlog2fc)
  )

ranks_df_D2D3 <- ranks_df_D2D3 %>%
  group_by(UNIPROT) %>%
  slice_max(order_by = abs(meanlog2fc), n = 1, with_ties = FALSE) %>%
  ungroup()

# D1 vs D3

fc_df_D1D3 <- df_log2 %>%
  filter(Timepoint %in% c("D1", "D3")) %>%
  pivot_longer(
    cols = -c(SubjectID, SampleId, Timepoint),
    names_to = "Protein",
    values_to = "log2_expr"
  ) %>%
  pivot_wider(
    id_cols    = c(SubjectID, Protein),
    names_from = Timepoint,
    values_from = log2_expr
  ) %>%
  filter(!is.na(D1) & !is.na(D3)) %>%
  mutate(log2_FC_D3_vs_D1 = D3 - D1)

ranks_df_D1D3 <- fc_df_D1D3 %>%
  group_by(Protein) %>%
  summarize(meanlog2fc = mean(log2_FC_D3_vs_D1, na.rm = TRUE), .groups = "drop")

ranks_df_D1D3 <- ranks_df_D1D3 %>%
  mutate(
    UNIPROT_raw = Protein,
    UNIPROT = str_replace(Protein, "\\.\\d+$", ""),   # removes trailing .<digits>
    meanlog2fc = as.numeric(meanlog2fc)
  )

ranks_df_D1D3 <- ranks_df_D1D3 %>%
  group_by(UNIPROT) %>%
  slice_max(order_by = abs(meanlog2fc), n = 1, with_ties = FALSE) %>%
  ungroup()

```

### convert from uniprot to entrez and create ranked vector

```{r}
library(org.Hs.eg.db)
library(AnnotationDbi)

map <- AnnotationDbi::select(
  org.Hs.eg.db,
  keys     = unique(ranks_df_D1D2$UNIPROT),
  keytype  = "UNIPROT",
  columns  = c("ENTREZID")
) %>%
  filter(!is.na(ENTREZID)) %>%
  distinct(UNIPROT, ENTREZID)

#Join mapping; if multiple UniProt map to the same Entrez, collapse to 1 per Entrez
ranks_entrez_df_D1D2 <- ranks_df_D1D2 %>%
  inner_join(map, by = "UNIPROT") %>%
  group_by(ENTREZID) %>%
  slice_max(order_by = abs(meanlog2fc), n = 1, with_ties = FALSE) %>%
  ungroup()

ranks_entrez_df_D2D3 <- ranks_df_D2D3 %>%
  inner_join(map, by = "UNIPROT") %>%
  group_by(ENTREZID) %>%
  slice_max(order_by = abs(meanlog2fc), n = 1, with_ties = FALSE) %>%
  ungroup()

ranks_entrez_df_D1D3 <- ranks_df_D1D3 %>%
  inner_join(map, by = "UNIPROT") %>%
  group_by(ENTREZID) %>%
  slice_max(order_by = abs(meanlog2fc), n = 1, with_ties = FALSE) %>%
  ungroup()

# Create fgsea-ready named vector
ranks_entrez_D1D2 <- ranks_entrez_df_D1D2$meanlog2fc
names(ranks_entrez_D1D2) <- ranks_entrez_df_D1D2$ENTREZID
ranks_entrez_D1D2 <- sort(ranks_entrez_D1D2, decreasing = TRUE)

ranks_entrez_D2D3 <- ranks_entrez_df_D2D3$meanlog2fc
names(ranks_entrez_D2D3) <- ranks_entrez_df_D2D3$ENTREZID
ranks_entrez_D2D3 <- sort(ranks_entrez_D2D3, decreasing = TRUE)

ranks_entrez_D1D3 <- ranks_entrez_df_D1D3$meanlog2fc
names(ranks_entrez_D1D3) <- ranks_entrez_df_D1D3$ENTREZID
ranks_entrez_D1D3 <- sort(ranks_entrez_D1D3, decreasing = TRUE)
```

### Run GO:BP

```{r}
library(clusterProfiler)

gsea_bp <- gseGO(
  geneList      = ranks_entrez_D1D2,
  OrgDb         = org.Hs.eg.db,
  keyType       = "ENTREZID",
  ont           = "BP",
  minGSSize     = 4,
  maxGSSize     = 800,
  pvalueCutoff  = 0.05,
  pAdjustMethod = "none",
)

head(as.data.frame(gsea_bp), 20)

library(enrichplot)

dotplotD1D2 <- dotplot(gsea_bp, showCategory = 7, split = ".sign") + facet_grid(.~.sign)
ridgeplotD1D2 <- ridgeplot(gsea_bp, showCategory = 10)

# Enrichment curve for a specific term (pick one from gsea_bp@result$ID)
enrichplotD1D2 <- gseaplot2(gsea_bp, geneSetID = gsea_bp@result$ID[1])

```

### Draw 2 vs Draw 3

```{r}
gsea_bp_d2d3 <- gseGO(
  geneList      = ranks_entrez_D2D3,
  OrgDb         = org.Hs.eg.db,
  keyType       = "ENTREZID",
  ont           = "BP",
  minGSSize     = 4,
  maxGSSize     = 800,
  pvalueCutoff  = 0.05,
  pAdjustMethod = "none",
)

head(as.data.frame(gsea_bp_d2d3), 20)

library(enrichplot)

dotplotd2d3 <- dotplot(gsea_bp_d2d3, showCategory = 7, split = ".sign") + facet_grid(.~.sign)
ridgeplotd2d3 <- ridgeplot(gsea_bp_d2d3, showCategory = 10)

# Enrichment curve for a specific term (pick one from gsea_bp@result$ID)
enrichplotd2d3 <- gseaplot2(gsea_bp_d2d3, geneSetID = gsea_bp@result$ID[1])
```

### Draw 1 vs Draw 3

```{r}
gsea_bp_d1d3 <- gseGO(
  geneList      = ranks_entrez_D1D3,
  OrgDb         = org.Hs.eg.db,
  keyType       = "ENTREZID",
  ont           = "BP",
  minGSSize     = 4,
  maxGSSize     = 800,
  pvalueCutoff  = 0.05,
  pAdjustMethod = "none",
)

head(as.data.frame(gsea_bp_d1d3), 20)

library(enrichplot)

dotplotd1d3 <- dotplot(gsea_bp_d1d3, showCategory = 7, split = ".sign") + facet_grid(.~.sign)
ridgeplotd1d3 <- ridgeplot(gsea_bp_d1d3, showCategory = 10)

# Enrichment curve for a specific term (pick one from gsea_bp@result$ID)
enrichplotd1d3 <- gseaplot2(gsea_bp_d1d3, geneSetID = gsea_bp@result$ID[1])
```

## Over-representation Analysis

-   use the stable features only from sPLSDA bootstrapping analysis
-   compare this to the list of all genes we measured, NOT THE WHOLE GENOME

### pull stable features from components 1 and 2 and convert to entrezID

```{r}
library(AnnotationDbi)
library(org.Hs.eg.db)

# Pull the STABLE features from each component for ORA.

# Step 1. stable features for component 1 (freq > 0.4)
stable_c1 <- stable_vars_04$Comp1
stable_c2 <- stable_vars_04$Comp2

# Step 2. pull the loadings and variable names for all 75 variables in component 1.

c1_loadings <- plsda_result$loadings$X[,1]

# Step 4. Split them into the positive and negative components for interpretability down the line (to know which pathways are up and which are down)

c1_pos_uniprot_ids <- names(c1_loadings)[c1_loadings > 0]
c1_neg_uniprot_ids <- names(c1_loadings)[c1_loadings < 0]


# Step 5. compare the uniprot IDs to the list of stable features, then remove the suffixes.
c1_pos_uniprot_ids_stable <- intersect(c1_pos_uniprot_ids, stable_c1)
c1_neg_uniprot_ids_stable <- intersect(c1_neg_uniprot_ids, stable_c1)

c1_pos_uniprot_ids_stable <- c1_pos_uniprot_ids_stable %>% str_replace("\\.\\d+$","")
c1_neg_uniprot_ids <- c1_neg_uniprot_ids_stable %>% str_replace("\\.\\d+$","")

# Map UniProt -> Entrez
pls_pos_map <- AnnotationDbi::select(
  org.Hs.eg.db,
  keys = unique(c1_pos_uniprot_ids_stable),
  keytype = "UNIPROT",
  columns = c("ENTREZID")
) %>%
  filter(!is.na(ENTREZID)) %>%
  distinct(UNIPROT, ENTREZID)

c1_pos_gene_ids <- unique(pls_pos_map$ENTREZID)

pls_neg_map <- AnnotationDbi::select(
  org.Hs.eg.db,
  keys = unique(c1_neg_uniprot_ids_stable),
  keytype = "UNIPROT",
  columns = c("ENTREZID")
) %>%
  filter(!is.na(ENTREZID)) %>%
  distinct(UNIPROT, ENTREZID)

c1_neg_gene_ids <- unique(pls_neg_map$ENTREZID)

# Repeat for component 2

c2_loadings <- plsda_result$loadings$X[,2]
c2_pos_uniprot_ids <- names(c2_loadings)[c2_loadings > 0]
c2_neg_uniprot_ids <- names(c2_loadings)[c2_loadings < 0]

c2_pos_uniprot_ids_stable <- intersect(c2_pos_uniprot_ids, stable_c2)
c2_neg_uniprot_ids_stable <- intersect(c2_neg_uniprot_ids, stable_c2)

c2_pos_uniprot_ids_stable <- c2_pos_uniprot_ids_stable %>% str_replace("\\.\\d+$","")
c2_neg_uniprot_ids_stable <- c2_neg_uniprot_ids_stable %>% str_replace("\\.\\d+$","")

pls_pos_map2 <- AnnotationDbi::select(
  org.Hs.eg.db,
  keys = unique(c2_pos_uniprot_ids_stable),
  keytype = "UNIPROT",
  columns = c("ENTREZID")
) %>%
  filter(!is.na(ENTREZID)) %>%
  distinct(UNIPROT, ENTREZID)

c2_pos_gene_ids <- unique(pls_pos_map2$ENTREZID)

pls_neg_map2 <- AnnotationDbi::select(
  org.Hs.eg.db,
  keys = unique(c2_neg_uniprot_ids_stable),
  keytype = "UNIPROT",
  columns = c("ENTREZID")
) %>%
  filter(!is.na(ENTREZID)) %>%
  distinct(UNIPROT, ENTREZID)

c2_neg_gene_ids <- unique(pls_neg_map2$ENTREZID)

```

### define the background "universe" as all proteins from our panel

```{r}
all_uniprot <- names(plsda_result$loadings$X[,1])
all_uniprot <- all_uniprot %>% str_replace("\\.\\d+$","")


bg_map <- AnnotationDbi::select(
  org.Hs.eg.db,
  keys = unique(all_uniprot),
  keytype = "UNIPROT",
  columns = c("ENTREZID")
) %>%
  filter(!is.na(ENTREZID)) %>%
  distinct(UNIPROT, ENTREZID)

all_entrez <- unique(bg_map$ENTREZID)
```


### run ORA with GO:BP or other desired databases
```{r}
library(clusterProfiler)
library(ReactomePA)
library(AnnotationDbi)
library(org.Hs.eg.db)

# ~~~~~~~~~~~~~~~~~~~~~~~~ for component 1 ~~~~~~~~~~~~~~~~~~~~~~~~

# ---GO:BP---

ora_bp_c1_pos <- enrichGO(
  gene          = c1_pos_gene_ids,
  universe      = all_entrez,          # all proteins in our panel
  OrgDb         = org.Hs.eg.db,
  keyType       = "ENTREZID",
  ont           = "BP",
  pAdjustMethod = "none",
  pvalueCutoff  = 0.05,
  qvalueCutoff = 1.0,
  minGSSize = 3,
  maxGSSize = 5000,
  readable      = TRUE
)

dotplotc1_pos <- enrichplot::dotplot(ora_bp_c1_pos, showCategory = 10)

ora_bp_c1_neg <- enrichGO(
  gene          = c1_neg_gene_ids,
  universe      = all_entrez,
  OrgDb         = org.Hs.eg.db,
  keyType       = "ENTREZID",
  ont           = "BP",
  pAdjustMethod = "none",
  pvalueCutoff  = 0.05,
  qvalueCutoff = 1.0,
  minGSSize = 3,
  maxGSSize = 5000,
  readable      = TRUE
)

dotplotc1_neg <- enrichplot::dotplot(ora_bp_c1_neg, showCategory = 10)

# ---Reactome---

ora_ra_c1_pos <- enrichPathway(
  gene = c1_pos_gene_ids,
  universe = all_entrez,
  organism = "human",
  pvalueCutoff = 0.05,
  pAdjustMethod = "none",
  qvalueCutoff = 1,
  readable = TRUE
)

dotplotRAc1_pos<- enrichplot::dotplot(ora_ra_c1_pos, showCategory = 10)

ora_ra_c1_neg <- enrichPathway(
  gene = c1_neg_gene_ids,
  universe = all_entrez,
  organism = "human",
  pvalueCutoff = 0.05,
  pAdjustMethod = "none",
  qvalueCutoff = 1,
  readable = TRUE
)

dotplotRAc1_neg <- enrichplot::dotplot(ora_ra_c1_neg, showCategory = 10)

# ~~~~~~~~~~~~~~~~~~~~~~`for component 2~~~~~~~~~~~~~~~~~~~~~~~~

# ---GO:BP---

#positive loadings
ora_bp_c2_pos <- enrichGO(
  gene          = c2_pos_gene_ids,
  universe      = all_entrez,
  OrgDb         = org.Hs.eg.db,
  keyType       = "ENTREZID",
  ont           = "BP",
  pAdjustMethod = "none",
  pvalueCutoff  = 0.05,
  qvalueCutoff = 1.0,
  minGSSize = 3,
  maxGSSize = 5000,
  readable      = TRUE               
)

dotplotc2_pos <- dotplot(ora_bp_c2_pos, showCategory = 10)

# negative loadings
ora_bp_c2_neg <- enrichGO(
  gene          = c2_neg_gene_ids,
  universe      = all_entrez,
  OrgDb         = org.Hs.eg.db,
  keyType       = "ENTREZID",
  ont           = "BP",
  pAdjustMethod = "none",
  pvalueCutoff  = 0.05,
  qvalueCutoff = 1.0,
  minGSSize = 3,
  maxGSSize = 5000,
  readable      = TRUE               
)

dotplotc2_neg <- dotplot(ora_bp_c2_neg, showCategory = 10)

# ---Reactome---
# positive loadings

ora_ra_c2_pos <- enrichPathway(
  gene = c2_pos_gene_ids,
  universe = all_entrez,
  organism = "human",
  pvalueCutoff = 0.05,
  pAdjustMethod = "none",
  qvalueCutoff = 1,
  readable = TRUE
)

dotplotRAc2_pos <- dotplot(ora_ra_c2_pos, showCategory = 10)

# negative loadings
ora_ra_c2_neg <- enrichPathway(
  gene = c2_neg_gene_ids,
  universe = all_entrez,
  organism = "human",
  pvalueCutoff = 0.05,
  pAdjustMethod = "none",
  qvalueCutoff = 1,
  readable = TRUE
)

dotplotRAc2_neg <- dotplot(ora_ra_c2_neg, showCategory = 10)
```

